mq/plugins/remove_dots.py

```diff
@@ -0,0 +1,38 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+# Copyright (c) 2014 Mozilla Corporation
+
+
+class message(object):
+
+    def __init__(self):
+        '''
+        takes an incoming message
+        and checks for dots at the
+        start or end of the key and
+        removes them
+        '''
+
+        self.registration = ['cloudtrail']
+        self.priority = 5
+
+    def onMessage(self, message, metadata):
+        def renameKeysToRemoveDots(message):
+            if isinstance(message, dict):
+                message_keys = list(message.keys())
+                for key in message_keys:
+                    print(key)
+                    if key[0] == '.' or key[-1] == '.':
+                        new_key = key.replace(".","")
+                        if new_key != key:
+                            message[new_key] = message.pop(key)
+                    if isinstance(message.get(key), dict) or isinstance(message.get(key), list):
+                        message[key] = renameKeysToRemoveDots(message[key])
+            elif isinstance(message, list):
+                for item in message:
+                    item = renameKeysToRemoveDots(item)
```

Comment:
There should be test cases for handling of all types of collections we deal with here.  This code, for example, would do nothing to modify message.  Example:

```py
>>> x = [".test", "test."]
>>>
>>> for item in x:
...   item = item.replace('.', '')
...
>>>
>>> x
['.test', 'test.']
```

Reply:
I'll add a list to the test cases :)  it does work, contrary to what your snippet shows.

Reply:
So I dug into this a little more, and while it's true, list items are not keys, so they aren't really subject to this thing, it did get me to reviewing the code a bit more. I did add a test case for a key that starts a list and has dots at  the start and end of the key. This works as expected.

However, for dictionaries within a list, it will not work for those, and that is quite alright, because any list is going to be indexed as an object, which means a nested dict within a list will be a part of that object, and will not be parsed into a key value pair.